src/models/base.tsx

```
@@ -85,40 +85,46 @@ export class Base {
+    if (!dims || dims.length < 3 || !dims[1] || !dims[2]) {
+      throw new Error('Invalid tensor dimensions');
+    }
+
+    const start = dims[2] * (dims[1] - 1);
     let max = arr[start];
     let maxidx = 0;
 
-    for (let i = 0; i < t.dims[2]; i++) {
+    for (let i = 0; i < dims[2]; i++) {
       const val = arr[i + start];
       if (!isFinite(val as number)) {
```

Biome (1.9.4)

[error] isFinite is unsafe. It attempts a type coercion. Use Number.isFinite instead.

See the MDN documentation for more details.
Unsafe fix: Use Number.isFinite instead.


(lint/suspicious/noGlobalIsFinite)